core/src/main/java/org/elasticsearch/common/unit/SizeValue.java

```diff
@@ -215,4 +234,29 @@ public int hashCode() {
         result = 31 * result + (sizeUnit != null ? sizeUnit.hashCode() : 0);
         return result;
     }
+
+    @Override
+    public int compareTo(SizeValue other) {
+        long unitSize = sizeUnit.toSingles(1);
+        long otherUnitSize = other.sizeUnit.toSingles(1);
+
+        SizeUnit minUnit = unitSize < otherUnitSize ? sizeUnit : other.sizeUnit;
+        long thisSize = this.convert(minUnit);
+        long otherSize = other.convert(minUnit);
+
+        if (thisSize == otherSize) {
+            if (thisSize == Long.MAX_VALUE || thisSize == Long.MIN_VALUE) {
```

Comment:
Same comment as the last class - I'd just convert to `SINGLE` here.

The equals method is funny on this class - I'd change it so it lines converts to `SINGLE` as well.